### PR TITLE
ARROW-17674: [R] Implement dplyr::across() inside arrange()

### PR DESCRIPTION
**Pull request by eitsupi**

### CODE REVIEW

File: r/R/dplyr-arrange.R [hunk 1]
```diff
@@ -20,7 +20,10 @@
 
 arrange.arrow_dplyr_query <- function(.data, ..., .by_group = FALSE) {
   call <- match.call()
-  exprs <- quos(...)
+
+  expression_list <- expand_across(.data, quos(...))
+  exprs <- ensure_named_exprs(expression_list)
```
Comment (nealrichardson) [inline r/R/dplyr-arrange.R:25]: Can this line be removed? I don't think arrange expressions have to have names.

**Review by nealrichardson — COMMENTED** [review]
Thanks for doing this! One question about a possible simplification

File: r/R/dplyr-arrange.R (hunk 1 above)
Comment (eitsupi) [reply r/R/dplyr-arrange.R]: Thank you for taking a look. I have removed the unnecessary lines.

**Review by nealrichardson — APPROVED** [review]
Thanks!

**Comment by eitsupi** [comment]
Thank you for merging this.
I hope you see #14090 and #14091 as they relate to this.

**Pull request merged**